src/appleseed/renderer/modeling/project/projectfileupdater.cpp

```diff
@@ -1786,6 +1786,60 @@ namespace
             params.remove_path("render_stamp_format");
         }
     };
+
+    //
+    // Update from revision 26 to revision 27.
+    //
+
+    class UpdateFromRevision_26
+      : public Updater
+    {
+      public:
+        explicit UpdateFromRevision_26(Project& project)
+          : Updater(project, 26)
+        {
+        }
+
+        void update() override
+        {
+            update_diagnostic_option();
+        }
+
+      private:
+        // Merge frame::save_extra_aovs and PixelRenderer::enable_diagnostics.
+        void update_diagnostic_option()
+        {
+            for (each<ConfigurationContainer> i = m_project.configurations(); i; ++i)
+            {
+                Dictionary& root = i->get_parameters();
+
+                if (root.dictionaries().exist("uniform_pixel_renderer"))
+                {
+                    Dictionary& upr = root.dictionary("uniform_pixel_renderer");
+
+                    if (upr.strings().exist("enable_diagnostics"))
+                        upr.strings().remove("enable_diagnostics");
```

Comment:
You can actually call `remove()` without first checking if the key exists; it won't do anything if the key doesn't exist.